Failed to remove X from collection null

**Describe the bug**
I cannot remove titles from the collection(maintainerr) by changing the rules and then re-running the rule.

**To Reproduce**
Steps to reproduce the behavior:
1. i created a rule
2. i ran the rule
3. checked the collection(maintainerr) and saw some titles i didn't want removed
4. changed the conditions for the rule
5. re-ran the rule
6. the collection(maintainerr) is unchanged and there are errors in the log

**Expected behavior**
I expect the collection(maintainerr) to be recreated with the new conditions

**Screenshots**

<img width="1285" height="663" alt="Image" src="https://github.com/user-attachments/assets/b5905f2d-5aba-4257-9fbc-e0e5db851d9c" />

<img width="331" height="309" alt="Image" src="https://github.com/user-attachments/assets/2ad48b4b-b916-4362-862c-2c0dc216d7e9" />

**Device (please complete the following information):**
 - OS: Unraid 7.2.4
 - Version 3.1.0

**Additional context**
The media server is Jellyfin, there is no collection created in jellyfin (and there shouldnt be since 'custom collection' is unchecked(?)).

logs
```
[maintainerr]  |  20/03/2026 19:29:04  [INFO] [CollectionsService] Couldn't remove media from collection: Required parameter collectionId was null or undefined when calling removeFromCollection.
[maintainerr]  |  20/03/2026 19:29:04  [INFO] [CollectionsService] Removing media with id a04e1eb5a93cc79a5ef93aca64d4a609 from collection..
[maintainerr]  |  20/03/2026 19:29:04  [ERROR] [JellyfinAdapterService] Failed to remove a04e1eb5a93cc79a5ef93aca64d4a609 from collection null
RequiredError: Required parameter collectionId was null or undefined when calling removeFromCollection.
```
repeated x44

i've ran the rule a couple times now, still getting the same errors.
im not sure why JellyfinAdapterService is running. its a bit confusing when both jellyfin and maintainerr both have "collections" meaning different things. as well as rules having rules instead of rules having conditions which to me feels more intuitive.

I asked this in discord and was told to post it in here instead.

--- a/apps/server/src/modules/collections/collections.service.ts
+++ b/apps/server/src/modules/collections/collections.service.ts
@@ -1156,46 +1156,48 @@ export class CollectionsService {
   }
 
   private async removeChildFromCollection(
-    collectionIds: { mediaServerId: string; dbId: number },
+    collectionIds: { mediaServerId: string | null; dbId: number },
     childId: string,
     logMeta?: CollectionLogMeta,
   ) {
     try {
-      const mediaServer = await this.getMediaServer();
       this.infoLogger(`Removing media with id ${childId} from collection..`);
 
-      try {
-        await mediaServer.removeFromCollection(
-          collectionIds.mediaServerId,
-          childId,
-        );
-
-        await this.connection
-          .createQueryBuilder()
-          .delete()
-          .from(CollectionMedia)
-          .where([
-            {
-              collectionId: collectionIds.dbId,
-              mediaServerId: childId,
-            },
-          ])
-          .execute();
-
-        await this.CollectionLogRecordForChild(
-          childId,
-          collectionIds.dbId,
-          'remove',
-          logMeta,
-        );
-      } catch (err) {
-        // 404 means media is not in collection, which is fine
-        if (!err.message?.includes('404')) {
-          this.infoLogger(
-            `Couldn't remove media from collection: ${err.message}`,
+      if (collectionIds.mediaServerId) {
+        try {
+          const mediaServer = await this.getMediaServer();
+          await mediaServer.removeFromCollection(
+            collectionIds.mediaServerId,
+            childId,
           );
+        } catch (err) {
+          // 404 means media is not in collection, which is fine
+          if (!err.message?.includes('404')) {
+            this.infoLogger(
+              `Couldn't remove media from collection: ${err.message}`,
+            );
+          }
         }
       }
+
+      await this.connection
+        .createQueryBuilder()
+        .delete()
+        .from(CollectionMedia)
+        .where([
+          {
+            collectionId: collectionIds.dbId,
+            mediaServerId: childId,
+          },
+        ])
+        .execute();
+
+      await this.CollectionLogRecordForChild(
+        childId,
+        collectionIds.dbId,
+        'remove',
+        logMeta,
+      );
     } catch (err) {
       this.logger.warn(
         'An error occurred while performing collection actions.',

